moveit_ros/planning/trajectory_execution_manager/src/trajectory_execution_manager.cpp

```diff
@@ -1312,7 +1313,23 @@ void TrajectoryExecutionManager::executeThread(const ExecutionCompleteCallback&
 
   // only report that execution finished successfully when the robot actually stopped moving
   if (last_execution_status_ == moveit_controller_manager::ExecutionStatus::SUCCEEDED)
-    waitForRobotToStop(*trajectories_[i - 1]);
+  {
+    execution_trajectories_mutex_.lock();
+    if (!execution_complete_ && trajectories_.size() >= i)
+    {
+      const auto& context = *trajectories_[i - 1];
+      std::vector<std::vector<std::string>> joint_names_per_traj;
+      for (const auto& trajectory : context.trajectory_parts_)
+      {
+        joint_names_per_traj.push_back(trajectory.joint_trajectory.joint_names);
+      }
+      execution_trajectories_mutex_.unlock();
+      waitForRobotToStop(joint_names_per_traj);
+    }
+    else
+      execution_trajectories_mutex_.unlock();
```

Comment:
Obsolete when using scoped_lock.
```suggestion
    else
      execution_trajectories_mutex_.unlock();
```